worlds/__init__.py

```diff
@@ -21,17 +24,23 @@
 lookup_any_location_id_to_name = {}
 games = {}
 
+# Build the datapackage for each game.
 for world_name, world in AutoWorldRegister.world_types.items():
     games[world_name] = {
-        "item_name_to_id" : world.item_name_to_id,
-        "location_name_to_id": world.location_name_to_id,
         "version": world.data_version,
+        "checksum": None,
+        "item_name_to_id": world.item_name_to_id,
+        "location_name_to_id": world.location_name_to_id,
         # seems clients don't actually want this. Keeping it here in case someone changes their mind.
         # "item_name_groups": {name: tuple(items) for name, items in world.item_name_groups.items()}
     }
     lookup_any_item_id_to_name.update(world.item_id_to_name)
     lookup_any_location_id_to_name.update(world.location_id_to_name)
 
+    # Calculate a checksum of the datapackage as a base64 encoded string.
+    checksum = base64.urlsafe_b64encode(hashlib.new("sha1", pickle.dumps(games[world_name])).digest()).decode()
+    games[world_name]["checksum"] = checksum.rstrip("=")
```

Comment:
As mentioned in discord, doing this lazily is probably better going forward. I would suggesting adding a get_data_package() or similar that caches in the class and construct the data package dynamically (see other comment above).

As background info, loading all worlds when launching an adjuster increases startup time quite a bit, and this is only getting worse. The steps to remedy this is
1. do as much as possible lazy
2. in the future, have to option to selectively import worlds, which probably breaks this for loop

We could do the change in two steps, accept this as-is, and clean up later, but I'd definitely like to get berserker's input since i think it's inevitable to change this sooner than later with our current growth rate.